src/turing_machine/machine.py

```diff
@@ -0,0 +1,41 @@
+class TuringMachine:
+    def __init__(self, tape, initial_state, final_states, transition_function):
+        self.tape = list(tape)
```

Comment:
Argument `tape` powinien być domyślnie listą, nie jest tu potrzebne już wywoływanie dodatkowo `list(tape)` ponieważ jeśli przekażemy w argumencie `tape` listę, to w tym miejscu otrzymamy listę list (czyli zagnieżdżoną listę)

Reply:
Tutaj cały czas, to samo. `tape` w argumencie funkcji jest listą, więc tutaj dodatkowe wywołanie `list(tape)` jest niepotrzebne, wystarczy `self.tape=tape`.